### ISSUE TITLE
Class 'Swagger\Client\Api\SignatureSellingPartner' not found

### ISSUE THREAD

**Issue by monsterUser** [clousale/amazon-sp-api-php#7]
Hello,

I try the FeedsApi example, but I get anytime this error:
Class 'Swagger\Client\Api\SignatureSellingPartner' not found

How can I fix this? I don't know what's wrong.

Thanks a lot!

`<?php

$config = Swagger\Client\Configuration::getDefaultConfiguration();
$config->setAccessToken('ZZZZZZZ'); //access token of Selling Partner
$config->setApiKey("accessKey", 'XXXXX'); // Access Key of IAM
$config->setApiKey("secretKey", 'YYYYY'); // Secret Key of IAM
$config->setApiKey("region", 'eu-west-1'); //region of MarketPlace country

$apiInstance = new Swagger\Client\Api\FeedsApi(
    // If you want use custom http client, pass your client which implements `GuzzleHttp\ClientInterface`.
    // This is optional, `GuzzleHttp\Client` will be used as default.
    new GuzzleHttp\Client(),
    $config
);
$body = new \Swagger\Client\Models\CreateFeedSpecification(); // \Swagger\Client\Models\CreateFeedSpecification | 

$apiInstance = new Swagger\Client\Api\CatalogApi(
	new GuzzleHttp\Client(),
	$config
);

?>`

That's my composer.json:

`{
    "require": {
        "guzzlehttp/guzzle": "^6.5",
        "zircote/swagger-php": "^3.1",
        "clousale/amazon-sp-api-php": "^1.1"
    }
}`

**Comment by cgaugel** [comment]
@monsterUser , you have to Rename the Class "Signature.php" to "SignatureSellingPartner.php" in /vendor/clousale/amazon-sp-api-php/lib

There are also some missing return statements in several Classes:

```
return new Request(
            'GET',
            $this->config->getHost() . $resourcePath . ($query ? "?{$query}" : ''),
            $headers,
            $httpBody
        );
```
Just for your information, if you need this in the future.

**Comment by monsterUser** [comment]
Thanks, now it works - but I get directly the next problem.

Now I get the error:
InvalidSignatureException

`$feed = "sku	product-id	product-id-type	price	item-condition	quantity	add-delete
TESTSKU	TESTASIN	1	300,00	11	1	a";

$body = new \Swagger\Client\Models\CreateFeedSpecification(); // \Swagger\Client\Models\CreateFeedSpecification | 
$body->setFeedType("_POST_FLAT_FILE_INVLOADER_DATA_");
$body->setMarketplaceIds("A1PA6795UKMFR9");
$body->setInputFeedDocumentId($feed);

echo "<pre>".print_r($body, true)."</pre>";

try {
    $result = $apiInstance->createFeed($body);
    echo "<pre>".print_r($result, true)."</pre>";
} catch (Exception $e) {
    echo 'Exception when calling FeedsApi->createFeed: ', $e->getMessage(), PHP_EOL;
}`

Is this definitly an problem with the keys and tokens? Or a code problem?

`

> POST /feeds/2020-09-04/feeds HTTP/1.1
--
  | Host: sellingpartnerapi-eu.amazon.com
  | Accept: application/json
  | Content-Type: application/json
  | Authorization: AWS4-HMAC-SHA256 Credential=YYYYYYYYYYYYYYYYYYYY/20201127/eu-west-1/execute-api/aws4_request, SignedHeaders=host;user-agent;x-amz-access-token;x-amz-date, Signature=f23cf00aaWWWWWWWWWWWWWWWWWW
  | user-agent: Swagger-Codegen/1.0.0/php
  | x-amz-access-token: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
  | x-amz-date: 20201127T161821Z
  | Content-Length: 243
  |  
  | * upload completely sent off: 243 out of 243 bytes
  | * Mark bundle as not supporting multiuse
  | < HTTP/1.1 403 Forbidden
  | < Date: Fri, 27 Nov 2020 16:18:21 GMT
  | < Content-Type: application/json
  | < Content-Length: 1101
  | < Connection: keep-alive
  | < x-amzn-RequestId: 123456-123456-123456
  | < x-amzn-ErrorType: InvalidSignatureException
  | < x-amz-apigw-id: ZZZZZZZZZZZZZZZZ
  | <
  | * Connection #0 to host sellingpartnerapi-eu.amazon.com left intact
  | Exception when calling FeedsApi->createFeed: [403] Client error: `POST https://sellingpartnerapi-eu.amazon.com/feeds/2020-09-04/feeds` resulted in a `403 Forbidden` response:
  | {
  | "errors": [
  | {
  | "message": "The request signature we calculated does not match the signature you provided. C (truncated...)

`

**Comment by cgaugel** [comment]
@monsterUser , do you have any news? I have nearly the same problem (see my Issue here: https://github.com/clousale/amazon-sp-api-php/issues/6)

**Comment by monsterUser** [comment]
@cgaugel Not yet! :( I don't know if something is wrong with my key's and token's or if it's a problem of the signature method or something else.

**Comment by vishalv26** [comment]
This library seems missing something very big please check "Request a Login with Amazon access token section" in https://github.com/amzn/selling-partner-api-docs/blob/main/guides/developer-guide/SellingPartnerApiDeveloperGuide.md#step-1-request-a-login-with-amazon-access-token

They say that the seller token must be exchanged with temporary access token and that access token should be included while signing the request.

**Comment by vishalv26** [comment]
A very quick dirty hack below to get token

```
$client = new GuzzleHttp\Client();

        $params = [
            'grant_type' => 'refresh_token',
            'refresh_token' => "#Your Long term Generated token",
            'client_id' => '#from seller central app',
            'client_secret' => '#from seller central app',
        ];
        $options = array_merge([
            RequestOptions::HEADERS     => ['Accept' => 'application/json'],
            RequestOptions::HTTP_ERRORS => false,
            'curl'                      => [CURLOPT_SSLVERSION => CURL_SSLVERSION_TLSv1_2],
        ], $params ? [RequestOptions::FORM_PARAMS => $params] : []);

        $response = $client->request('POST', 'https://api.amazon.com/auth/o2/token', $options);


$body = $response->getBody()->getContents();
```

**Comment by monsterUser** [comment]
I have the right token, thanks! My mainproblem at the mom is the hashedPayload - this is not correct - because $data is empty.

lib/Api/FeedsApi.php
Line 397

Last parameter is missing.

**Comment by clousale** [comment]
This is fixed with version 1.2 - please update!

**Comment by lozo55** [comment]
Hi, i have the same problem. :|
